sdk/core/azure-core/inc/azure/core/http/win_http_transport.hpp

```diff
@@ -140,8 +132,9 @@ namespace Azure { namespace Core { namespace Http {
   class WinHttpTransport : public HttpTransport {
   private:
     WinHttpTransportOptions m_options;
+    HINTERNET m_sessionHandle;
```

Comment:
a transport adapter can't have state because it would be non-thread safe. For example, with this change, we can create a WinHttpTransportAdapter and use it from 2 threads and have a rice of both threads creating the initial session at the same time.

If you want to keep sessions, you would need something lock-protected and I would expect to also consider the host/request configuration to check if there's a session to be used or if a new session is required for each request.

Reply:
> a transport adapter can't have state because it would be non-thread safe

It already has state, which is the options struct, (though atm, that is empty).

What if I I create the session within the ctor?

Reply:
I can make the session handle read only as well, by creating it once in the WinHttpTransport ctor.